### PR TITLE
Image Similarity PowerSkill using ResNet

### PR DESCRIPTION
**Pull request by shanepeckham**
## Purpose

* ...
This PowerSkill uses ResNet to return the TopN most similar images
## Does this introduce a breaking change?

```
[ ] Yes
[ X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
The code can be tested in entirety in the jupyter notebook "Similarity Search dogs"
```

## What to Check
Verify that the following are valid
* Run all the cells in the  jupyter notebook "Similarity Search dogs" and confirm the results are as documented in both the README and jupyter notebook

### CODE REVIEW

File: Vision/ImageSimilarity/deployment/azuresearch/create_skillset.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                }
+            ]
+        }
+    ],
+    "cognitiveServices": {
+        "@odata.type": "#Microsoft.Azure.Search.CognitiveServicesByKey",
+        "description": "cogsvc",
+        "key": ""
+    }
+}
```
Comment (vkurpad) [inline Vision/ImageSimilarity/deployment/azuresearch/create_skillset.json:68]: Can we add a knowledge store projection of the extracted images as a way to get the training dataset setup?

File: README.md [hunk 2]
```diff
@@ -23,6 +23,7 @@ This project provides the following custom skills:
 * [**AnalyzeForm**](Vision/AnalyzeForm/README.md): recognizes form fields in a document.
 * [**CustomVision**](Vision/CustomVision/README.md): classifies documents using [Custom Vision](https://customvision.ai) models.
 * [**CustomVision**](Vision/ImageClusteringSkill/README.md): Uses clustering to automatically group and label images
+* [**CustomVision**](Vision/ImageSimilarity/README.md): Uses ResNet to find the topn most similar images
```
Comment (vkurpad) [inline README.md:26]: top-n

**Review by vkurpad — COMMENTED** [review]
I have not tried deploying and running it yet, but it looks good!. Thanks for the PR

File: Vision/ImageSimilarity/deployment/azuresearch/create_skillset.json (hunk 1 above)
Comment (shanepeckham) [reply Vision/ImageSimilarity/deployment/azuresearch/create_skillset.json:68]: Added

File: README.md (hunk 2 above)
Comment (shanepeckham) [reply README.md]: Corrected

**Comment by bleroy** [comment]
Thanks for the contribution! Can you remove the empty dogs.zip file? What's in the mountain_dog4.txt file? I don't see that used anywhere.

**Comment by shanepeckham** [comment]
> Thanks for the contribution! Can you remove the empty dogs.zip file? What's in the mountain_dog4.txt file? I don't see that used anywhere.

@bleroy , the dogs.zip file contains the data for this PowerSkill. The mountain_dogs4.txt is used to test the API in the notebook Similarity Search dogs.ipynb

**Comment by bleroy** [comment]
But it's empty...
I checked the base64 txt file is indeed a picture of a dog.

**Comment by shanepeckham** [comment]
> But it's empty...
> I checked the base64 txt file is indeed a picture of a dog.

@bleroy  - I am a bit confused, I have just downloaded dogs.zip and extracted 18 images from it. Hmmm

**Comment by bleroy** [comment]
I guess gitHub is somehow confused about it: 
image

Oh well, you're right, sorry about that.

File: Vision/ImageSimilarity/README.md [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+## High level Process
+
+![image similarity process](images/image_similarity_flow.png)
+
+## How to implement
+
+This section describes how to get this working on sample data
+and how it can be amended for your data.
+ 
+1) ###Data
```
Comment (bleroy) [inline Vision/ImageSimilarity/README.md:40]: The title markers `###` don't seem to work with numbered lists, at least on GH:

image

Comment (shanepeckham) [reply Vision/ImageSimilarity/README.md]: @bleroy - Fixed the markdown

**Comment by bleroy** [comment]
Thanks for the contribution!

**Pull request merged**